gossip/infostore.go

```diff
@@ -274,24 +283,29 @@ func (is *infoStore) combine(delta *infoStore) int {
 // from node requesting delta are ignored.
 //
 // Returns nil if there are no deltas.
-func (is *infoStore) delta(nodeID proto.NodeID, seq int64) *infoStore {
-	if seq >= is.MaxSeq {
-		return nil
+func (is *infoStore) delta(nodeID proto.NodeID, seq int64) InfoStoreDelta {
+	delta := InfoStoreDelta{
+		NodeID:   is.NodeID,
+		NodeAddr: is.NodeAddr,
+		MaxSeq:   is.MaxSeq,
 	}
 
-	delta := newInfoStore(is.NodeID, is.NodeAddr)
+	if seq < is.MaxSeq {
+		delta.Infos = make(map[string]*Info)
 
-	// Compute delta of infos.
-	if err := is.visitInfos(func(key string, i *info) error {
-		if i.isFresh(nodeID, seq) {
-			return delta.addInfo(key, i)
-		}
-		return nil
-	}); err != nil {
-		log.Errorf("failed to properly create delta infoStore: %s", err)
+		// Compute delta of infos.
+		// will never error because `return nil` below
```

Comment:
on master: `git grep -F 'will never error because `return nil` below' | wc -l` -> 8

I'm not too keen on changing them all.